lib/telephone_number/formatter.rb

```diff
@@ -1,111 +1,76 @@
 module TelephoneNumber
   class Formatter
-    extend Forwardable
 
-    attr_reader :normalized_number, :phone_data, :valid
+    attr_reader :normalized_number, :country, :valid, :original_number
 
-    delegate [:country_data, :country] => :phone_data
-
-    def initialize(number_obj, phone_data)
+    def initialize(number_obj)
       @normalized_number = number_obj.normalized_number
-      @phone_data = phone_data
+      @country = number_obj.country
       @valid = number_obj.valid?
+      @original_number = number_obj.original_number
     end
 
     def national_number(formatted: true)
-      if formatted
-        @formatted_national_number ||= build_national_number
-      else
-        @national_number ||= build_national_number(formatted: false)
-      end
+      return sanitized_or_default if !valid? || number_format.nil?
+      build_national_number(formatted: formatted)
     end
 
     def e164_number(formatted: true)
-      if formatted
-        @formatted_e164_number ||= build_e164_number
-      else
-        @e164_number ||= build_e164_number(formatted: false)
-      end
+      return sanitized_or_default if !valid
```

Comment:
No, you're correct. I changed this, because it seems a little odd.  I mean if a number is valid, then it *should* have a corresponding format. Great. But if it's not, should we try to find a format anyways? Or just say "nah man, you'll get a prettily formatted number when you give us a valid number"?